### ISSUE TITLE
Failing dependency resolution with hex

### ISSUE THREAD

**Issue by GalaxyGorilla** [elixir-lang/elixir#3872]
I have a small elixir app with lager as the only dependency, e.g.

  defp deps do
    [{:lager, "~> 2.1.1"}]
  end

Trying to compile this i get the error

  ==> lager (compile)
  Dependency not available: goldrush-0.1.6 ({git,
                                             "git://github.com/DeadZen/goldrush.git",
                                             {tag,"0.1.6"}})
  ** (Mix) Could not compile dependency :lager, "/usr/bin/rebar" command failed. You can recompile this dependency with "mix deps.compile lager", update it with "mix deps.update lager" or clean it with "mix deps.clean lager"

because hex fetches the goldrush package with version 0.1.7 instead of the git repo with version 0.1.6. Using 'mix deps.get' this is the relevant output:

  * Getting goldrush (Hex package) 
    Checking package (https://s3.amazonaws.com/s3.hex.pm/tarballs/goldrush-0.1.7.tar)

I'm not sure if this is a bug or I'm doing something completely wrong. I already cleaned the caches but that was not successful.

**Comment by surik** [comment]
hex_registry ets has this:
```erlang
{{<<"lager">>,<<"2.1.1">>},
 [[[<<"goldrush">>,<<"~> 0.1.6">>,false,<<"goldrush">>]],
  <<"8E389F916384ABF5F62169969BD7515C4C252CBBDA53FEEBBEF90F47DB5EAAD1">>,
  [<<"make">>,<<"rebar">>]]}
```
and it fetches goldrush 0.1.7 because it is available but lager wants 0.1.6 in rebar.config

**Comment by josevalim** [comment]
You should report it to rebar. Whoever published rebar 2.1 used the wrong dependency specification: https://hex.pm/packages/lager/2.1.1

You can see it says `goldrush ~> 0.1.6` while the config requires: https://github.com/basho/lager/blob/2.1.1/rebar.config#L5

Alternatively, you can just specify the proper goldrush in your app. :)
